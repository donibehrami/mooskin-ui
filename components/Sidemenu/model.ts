import { IBoxComponentProps } from '../Box/model';

export interface ISidemenuComponentProps extends IBoxComponentProps {
    /** active Sidemenu item */
    activeItem?: string;

    /** callback called when clicking a Sidemenu item */
    onClickItem?: (e: React.MouseEvent<HTMLElement>, value?: string) => void;
}

export interface ISidemenuItemComponentProps extends IBoxComponentProps {
    /** value returned in callback fn */
    value?: string;

    /** whether the item is active */
    active?: boolean;
}
