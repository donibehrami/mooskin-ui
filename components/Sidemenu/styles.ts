import styled from 'styled-components';

// Models
import { ISidemenuComponentProps, ISidemenuItemComponentProps } from './model';

// Components
import Box from '../Box/Box';

// import variables from '../_utils/globals/variables';

export const StyledSidemenu = styled(Box)<ISidemenuComponentProps>`
    border-left: 1px solid #e2e2e2;
    display: flex;
    flex-direction: column;
`;

export const StyledSidemenuItem = styled(Box)<ISidemenuItemComponentProps>`
    font-family: Montserrat;
    font-size: 14px;
    font-weight: ${(props) => props.active ? 'bold' : '500'};
    font-stretch: normal;
    font-style: normal;
    line-height: 1.29;
    letter-spacing: normal;
    text-align: left;
    color: #293346;
    margin-left: ${(props) => props.active ? '-2px' : '0'};
    padding-left: 9px;
    border-left: ${(props) => props.active ? '3px solid #3fbaca' : ''};
    transition: 0.3s all ease;
    cursor: pointer;
    :not(:last-child){
        margin-bottom: 32px;
    }
`;
