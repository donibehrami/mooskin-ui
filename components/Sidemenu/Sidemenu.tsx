import * as React from 'react';

// Models
import { ISidemenuComponentProps, ISidemenuItemComponentProps } from './model';

// Styled Components
import { StyledSidemenu, StyledSidemenuItem } from './styles';

/**
 * Sidemenu
 */
export const Sidemenu: React.FC<ISidemenuComponentProps> = (props) => {
    const batchClickHandler = (
        e: React.MouseEvent<HTMLElement>,
        value?: string,
        callback?: (e: React.MouseEvent<HTMLElement>, value?: string) => void
    ) => {
        props.onClickItem && props.onClickItem(e, value);
        callback && callback(e, value);
    };

    const recurseChildren = (children: any): any => {
        return React.Children.map(children, (child, i) => {
            if (React.isValidElement<ISidemenuItemComponentProps>(child) && child.type === SidemenuItem) {
                return React.cloneElement(child, {
                    active: child.props.value === props.activeItem,
                    children: recurseChildren((child.props as any).children),
                    key: i,
                    onClick: (e) => batchClickHandler(e, child.props.value, child.props.onClick)
                } as ISidemenuItemComponentProps);
            }

            if (React.isValidElement(child) && (child.props as any).children) {
                return React.cloneElement(child, { key: i, children: recurseChildren((child.props as any).children) } as any);
            }

            return child;
        });
    };

    return <StyledSidemenu {...props} children={recurseChildren(props.children)} />;
};

Sidemenu.defaultProps = {
    className: '',
    style: {}
};

/**
 * SidemenuItem
 */
export const SidemenuItem: React.FC<ISidemenuItemComponentProps> = (props) => {
    return <StyledSidemenuItem {...props} />;
};

SidemenuItem.defaultProps = {
    className: '',
    style: {}
};

export default Sidemenu;
