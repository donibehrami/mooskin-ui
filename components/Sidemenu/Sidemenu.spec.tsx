import * as React from 'react';

import Sidemenu, {SidemenuItem} from './Sidemenu';

import { mount, shallow } from 'enzyme';

describe('Sidemenu', () => {

    test('renders Sidemenu correctly', () => {
        const func = jest.fn();

        const tree = shallow(
            <Sidemenu activeItem="/settings" onClickItem={func}>
                <SidemenuItem dataLabel="settings" value="/settings">
                    Settings
                </SidemenuItem>
                <SidemenuItem dataLabel="template" value="/template">
                    Template
                </SidemenuItem>
                <SidemenuItem dataLabel="preview" value="/preview">
                    Preview
                </SidemenuItem>
            </Sidemenu>
        );
        expect(tree).toMatchSnapshot();
    });

    test('renders SidemenuItem correctly', () => {
        const func = jest.fn();

        const tree = shallow(
            <SidemenuItem
                onClick={func}
                className="myClass"
                style={{color: 'blue'}}
                active
            />
        );
        expect(tree).toMatchSnapshot();
    });

    test('calls click callbacks correctly', () => {
        const func = jest.fn();

        const tree = mount(
            <Sidemenu activeItem="/settings" onClickItem={func}>
                <SidemenuItem dataLabel="settings" value="/settings">
                    Settings
                </SidemenuItem>
                <SidemenuItem dataLabel="template" value="/template">
                    Template
                </SidemenuItem>
                <SidemenuItem dataLabel="preview" value="/preview">
                    Preview
                </SidemenuItem>
            </Sidemenu>
        );

        tree.find('div').at(2).simulate('click');
        expect(func).toHaveBeenCalled();
    });

});
