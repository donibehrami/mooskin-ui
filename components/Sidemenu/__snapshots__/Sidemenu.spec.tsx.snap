// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidemenu renders Sidemenu correctly 1`] = `
<WithMooskinContext
  activeItem="/settings"
  className=""
  onClickItem={[MockFunction]}
  style={Object {}}
>
  <Component
    activeItem="/settings"
    className=""
    onClickItem={[MockFunction]}
    palette={
      Object {
        "backgroundColors": Object {
          "button": "#3fbaca",
          "common": "#ffffff",
          "disabledButton": "#e2e2e2",
          "main": "#f7f7f7",
          "secondaryToggle": "#ffffff",
          "toggle": "#425579",
        },
        "borderColors": Object {
          "common": "#e2e2e2",
        },
        "fontColors": Object {
          "anchor": "#3fbaca",
          "button": "#ffffff",
          "common": "#293346",
          "description": "#9d9d9d",
          "disabledAnchor": "#9d9d9d",
          "label": "#2d2d2d",
          "secondaryToggle": "#9d9d9d",
          "text": "#2d2d2d",
          "toggle": "#ffffff",
        },
      }
    }
    style={Object {}}
  >
    <StyledSidemenu
      activeItem="/settings"
      className=""
      onClickItem={[MockFunction]}
      palette={
        Object {
          "backgroundColors": Object {
            "button": "#3fbaca",
            "common": "#ffffff",
            "disabledButton": "#e2e2e2",
            "main": "#f7f7f7",
            "secondaryToggle": "#ffffff",
            "toggle": "#425579",
          },
          "borderColors": Object {
            "common": "#e2e2e2",
          },
          "fontColors": Object {
            "anchor": "#3fbaca",
            "button": "#ffffff",
            "common": "#293346",
            "description": "#9d9d9d",
            "disabledAnchor": "#9d9d9d",
            "label": "#2d2d2d",
            "secondaryToggle": "#9d9d9d",
            "text": "#2d2d2d",
            "toggle": "#ffffff",
          },
        }
      }
      style={Object {}}
    >
      <Box
        activeItem="/settings"
        className="sc-gtssRu gGAwqn"
        onClickItem={[MockFunction]}
        palette={
          Object {
            "backgroundColors": Object {
              "button": "#3fbaca",
              "common": "#ffffff",
              "disabledButton": "#e2e2e2",
              "main": "#f7f7f7",
              "secondaryToggle": "#ffffff",
              "toggle": "#425579",
            },
            "borderColors": Object {
              "common": "#e2e2e2",
            },
            "fontColors": Object {
              "anchor": "#3fbaca",
              "button": "#ffffff",
              "common": "#293346",
              "description": "#9d9d9d",
              "disabledAnchor": "#9d9d9d",
              "label": "#2d2d2d",
              "secondaryToggle": "#9d9d9d",
              "text": "#2d2d2d",
              "toggle": "#ffffff",
            },
          }
        }
        style={Object {}}
      >
        <Component
          activeItem="/settings"
          className="sc-gtssRu gGAwqn"
          onClickItem={[MockFunction]}
          palette={
            Object {
              "backgroundColors": Object {
                "button": "#3fbaca",
                "common": "#ffffff",
                "disabledButton": "#e2e2e2",
                "main": "#f7f7f7",
                "secondaryToggle": "#ffffff",
                "toggle": "#425579",
              },
              "borderColors": Object {
                "common": "#e2e2e2",
              },
              "fontColors": Object {
                "anchor": "#3fbaca",
                "button": "#ffffff",
                "common": "#293346",
                "description": "#9d9d9d",
                "disabledAnchor": "#9d9d9d",
                "label": "#2d2d2d",
                "secondaryToggle": "#9d9d9d",
                "text": "#2d2d2d",
                "toggle": "#ffffff",
              },
            }
          }
          style={Object {}}
        >
          <StyledBox
            activeItem="/settings"
            className="sc-gtssRu gGAwqn"
            onClickItem={[MockFunction]}
            palette={
              Object {
                "backgroundColors": Object {
                  "button": "#3fbaca",
                  "common": "#ffffff",
                  "disabledButton": "#e2e2e2",
                  "main": "#f7f7f7",
                  "secondaryToggle": "#ffffff",
                  "toggle": "#425579",
                },
                "borderColors": Object {
                  "common": "#e2e2e2",
                },
                "fontColors": Object {
                  "anchor": "#3fbaca",
                  "button": "#ffffff",
                  "common": "#293346",
                  "description": "#9d9d9d",
                  "disabledAnchor": "#9d9d9d",
                  "label": "#2d2d2d",
                  "secondaryToggle": "#9d9d9d",
                  "text": "#2d2d2d",
                  "toggle": "#ffffff",
                },
              }
            }
            style={Object {}}
          >
            <div
              className="sc-bdnylx sJSbd sc-gtssRu gGAwqn"
              onClickItem={[MockFunction]}
              style={Object {}}
            >
              <WithMooskinContext
                active={true}
                className=""
                key="0/.0"
                onClick={[Function]}
                style={Object {}}
                value="/settings"
              >
                <Component
                  active={true}
                  className=""
                  onClick={[Function]}
                  palette={
                    Object {
                      "backgroundColors": Object {
                        "button": "#3fbaca",
                        "common": "#ffffff",
                        "disabledButton": "#e2e2e2",
                        "main": "#f7f7f7",
                        "secondaryToggle": "#ffffff",
                        "toggle": "#425579",
                      },
                      "borderColors": Object {
                        "common": "#e2e2e2",
                      },
                      "fontColors": Object {
                        "anchor": "#3fbaca",
                        "button": "#ffffff",
                        "common": "#293346",
                        "description": "#9d9d9d",
                        "disabledAnchor": "#9d9d9d",
                        "label": "#2d2d2d",
                        "secondaryToggle": "#9d9d9d",
                        "text": "#2d2d2d",
                        "toggle": "#ffffff",
                      },
                    }
                  }
                  style={Object {}}
                  value="/settings"
                >
                  <StyledSidemenuItem
                    active={true}
                    className=""
                    onClick={[Function]}
                    palette={
                      Object {
                        "backgroundColors": Object {
                          "button": "#3fbaca",
                          "common": "#ffffff",
                          "disabledButton": "#e2e2e2",
                          "main": "#f7f7f7",
                          "secondaryToggle": "#ffffff",
                          "toggle": "#425579",
                        },
                        "borderColors": Object {
                          "common": "#e2e2e2",
                        },
                        "fontColors": Object {
                          "anchor": "#3fbaca",
                          "button": "#ffffff",
                          "common": "#293346",
                          "description": "#9d9d9d",
                          "disabledAnchor": "#9d9d9d",
                          "label": "#2d2d2d",
                          "secondaryToggle": "#9d9d9d",
                          "text": "#2d2d2d",
                          "toggle": "#ffffff",
                        },
                      }
                    }
                    style={Object {}}
                    value="/settings"
                  >
                    <Box
                      active={true}
                      className="sc-dlnjPT dhtinz"
                      onClick={[Function]}
                      palette={
                        Object {
                          "backgroundColors": Object {
                            "button": "#3fbaca",
                            "common": "#ffffff",
                            "disabledButton": "#e2e2e2",
                            "main": "#f7f7f7",
                            "secondaryToggle": "#ffffff",
                            "toggle": "#425579",
                          },
                          "borderColors": Object {
                            "common": "#e2e2e2",
                          },
                          "fontColors": Object {
                            "anchor": "#3fbaca",
                            "button": "#ffffff",
                            "common": "#293346",
                            "description": "#9d9d9d",
                            "disabledAnchor": "#9d9d9d",
                            "label": "#2d2d2d",
                            "secondaryToggle": "#9d9d9d",
                            "text": "#2d2d2d",
                            "toggle": "#ffffff",
                          },
                        }
                      }
                      style={Object {}}
                      value="/settings"
                    >
                      <Component
                        active={true}
                        className="sc-dlnjPT dhtinz"
                        onClick={[Function]}
                        palette={
                          Object {
                            "backgroundColors": Object {
                              "button": "#3fbaca",
                              "common": "#ffffff",
                              "disabledButton": "#e2e2e2",
                              "main": "#f7f7f7",
                              "secondaryToggle": "#ffffff",
                              "toggle": "#425579",
                            },
                            "borderColors": Object {
                              "common": "#e2e2e2",
                            },
                            "fontColors": Object {
                              "anchor": "#3fbaca",
                              "button": "#ffffff",
                              "common": "#293346",
                              "description": "#9d9d9d",
                              "disabledAnchor": "#9d9d9d",
                              "label": "#2d2d2d",
                              "secondaryToggle": "#9d9d9d",
                              "text": "#2d2d2d",
                              "toggle": "#ffffff",
                            },
                          }
                        }
                        style={Object {}}
                        value="/settings"
                      >
                        <StyledBox
                          active={true}
                          className="sc-dlnjPT dhtinz"
                          onClick={[Function]}
                          palette={
                            Object {
                              "backgroundColors": Object {
                                "button": "#3fbaca",
                                "common": "#ffffff",
                                "disabledButton": "#e2e2e2",
                                "main": "#f7f7f7",
                                "secondaryToggle": "#ffffff",
                                "toggle": "#425579",
                              },
                              "borderColors": Object {
                                "common": "#e2e2e2",
                              },
                              "fontColors": Object {
                                "anchor": "#3fbaca",
                                "button": "#ffffff",
                                "common": "#293346",
                                "description": "#9d9d9d",
                                "disabledAnchor": "#9d9d9d",
                                "label": "#2d2d2d",
                                "secondaryToggle": "#9d9d9d",
                                "text": "#2d2d2d",
                                "toggle": "#ffffff",
                              },
                            }
                          }
                          style={Object {}}
                          value="/settings"
                        >
                          <div
                            className="sc-bdnylx sJSbd sc-dlnjPT dhtinz"
                            onClick={[Function]}
                            style={Object {}}
                            value="/settings"
                          >
                            Settings
                          </div>
                        </StyledBox>
                      </Component>
                    </Box>
                  </StyledSidemenuItem>
                </Component>
              </WithMooskinContext>
              <WithMooskinContext
                active={false}
                className=""
                key="1/.1"
                onClick={[Function]}
                style={Object {}}
                value="/template"
              >
                <Component
                  active={false}
                  className=""
                  onClick={[Function]}
                  palette={
                    Object {
                      "backgroundColors": Object {
                        "button": "#3fbaca",
                        "common": "#ffffff",
                        "disabledButton": "#e2e2e2",
                        "main": "#f7f7f7",
                        "secondaryToggle": "#ffffff",
                        "toggle": "#425579",
                      },
                      "borderColors": Object {
                        "common": "#e2e2e2",
                      },
                      "fontColors": Object {
                        "anchor": "#3fbaca",
                        "button": "#ffffff",
                        "common": "#293346",
                        "description": "#9d9d9d",
                        "disabledAnchor": "#9d9d9d",
                        "label": "#2d2d2d",
                        "secondaryToggle": "#9d9d9d",
                        "text": "#2d2d2d",
                        "toggle": "#ffffff",
                      },
                    }
                  }
                  style={Object {}}
                  value="/template"
                >
                  <StyledSidemenuItem
                    active={false}
                    className=""
                    onClick={[Function]}
                    palette={
                      Object {
                        "backgroundColors": Object {
                          "button": "#3fbaca",
                          "common": "#ffffff",
                          "disabledButton": "#e2e2e2",
                          "main": "#f7f7f7",
                          "secondaryToggle": "#ffffff",
                          "toggle": "#425579",
                        },
                        "borderColors": Object {
                          "common": "#e2e2e2",
                        },
                        "fontColors": Object {
                          "anchor": "#3fbaca",
                          "button": "#ffffff",
                          "common": "#293346",
                          "description": "#9d9d9d",
                          "disabledAnchor": "#9d9d9d",
                          "label": "#2d2d2d",
                          "secondaryToggle": "#9d9d9d",
                          "text": "#2d2d2d",
                          "toggle": "#ffffff",
                        },
                      }
                    }
                    style={Object {}}
                    value="/template"
                  >
                    <Box
                      active={false}
                      className="sc-dlnjPT cnwfAJ"
                      onClick={[Function]}
                      palette={
                        Object {
                          "backgroundColors": Object {
                            "button": "#3fbaca",
                            "common": "#ffffff",
                            "disabledButton": "#e2e2e2",
                            "main": "#f7f7f7",
                            "secondaryToggle": "#ffffff",
                            "toggle": "#425579",
                          },
                          "borderColors": Object {
                            "common": "#e2e2e2",
                          },
                          "fontColors": Object {
                            "anchor": "#3fbaca",
                            "button": "#ffffff",
                            "common": "#293346",
                            "description": "#9d9d9d",
                            "disabledAnchor": "#9d9d9d",
                            "label": "#2d2d2d",
                            "secondaryToggle": "#9d9d9d",
                            "text": "#2d2d2d",
                            "toggle": "#ffffff",
                          },
                        }
                      }
                      style={Object {}}
                      value="/template"
                    >
                      <Component
                        active={false}
                        className="sc-dlnjPT cnwfAJ"
                        onClick={[Function]}
                        palette={
                          Object {
                            "backgroundColors": Object {
                              "button": "#3fbaca",
                              "common": "#ffffff",
                              "disabledButton": "#e2e2e2",
                              "main": "#f7f7f7",
                              "secondaryToggle": "#ffffff",
                              "toggle": "#425579",
                            },
                            "borderColors": Object {
                              "common": "#e2e2e2",
                            },
                            "fontColors": Object {
                              "anchor": "#3fbaca",
                              "button": "#ffffff",
                              "common": "#293346",
                              "description": "#9d9d9d",
                              "disabledAnchor": "#9d9d9d",
                              "label": "#2d2d2d",
                              "secondaryToggle": "#9d9d9d",
                              "text": "#2d2d2d",
                              "toggle": "#ffffff",
                            },
                          }
                        }
                        style={Object {}}
                        value="/template"
                      >
                        <StyledBox
                          active={false}
                          className="sc-dlnjPT cnwfAJ"
                          onClick={[Function]}
                          palette={
                            Object {
                              "backgroundColors": Object {
                                "button": "#3fbaca",
                                "common": "#ffffff",
                                "disabledButton": "#e2e2e2",
                                "main": "#f7f7f7",
                                "secondaryToggle": "#ffffff",
                                "toggle": "#425579",
                              },
                              "borderColors": Object {
                                "common": "#e2e2e2",
                              },
                              "fontColors": Object {
                                "anchor": "#3fbaca",
                                "button": "#ffffff",
                                "common": "#293346",
                                "description": "#9d9d9d",
                                "disabledAnchor": "#9d9d9d",
                                "label": "#2d2d2d",
                                "secondaryToggle": "#9d9d9d",
                                "text": "#2d2d2d",
                                "toggle": "#ffffff",
                              },
                            }
                          }
                          style={Object {}}
                          value="/template"
                        >
                          <div
                            className="sc-bdnylx sJSbd sc-dlnjPT cnwfAJ"
                            onClick={[Function]}
                            style={Object {}}
                            value="/template"
                          >
                            Template
                          </div>
                        </StyledBox>
                      </Component>
                    </Box>
                  </StyledSidemenuItem>
                </Component>
              </WithMooskinContext>
              <WithMooskinContext
                active={false}
                className=""
                key="2/.2"
                onClick={[Function]}
                style={Object {}}
                value="/preview"
              >
                <Component
                  active={false}
                  className=""
                  onClick={[Function]}
                  palette={
                    Object {
                      "backgroundColors": Object {
                        "button": "#3fbaca",
                        "common": "#ffffff",
                        "disabledButton": "#e2e2e2",
                        "main": "#f7f7f7",
                        "secondaryToggle": "#ffffff",
                        "toggle": "#425579",
                      },
                      "borderColors": Object {
                        "common": "#e2e2e2",
                      },
                      "fontColors": Object {
                        "anchor": "#3fbaca",
                        "button": "#ffffff",
                        "common": "#293346",
                        "description": "#9d9d9d",
                        "disabledAnchor": "#9d9d9d",
                        "label": "#2d2d2d",
                        "secondaryToggle": "#9d9d9d",
                        "text": "#2d2d2d",
                        "toggle": "#ffffff",
                      },
                    }
                  }
                  style={Object {}}
                  value="/preview"
                >
                  <StyledSidemenuItem
                    active={false}
                    className=""
                    onClick={[Function]}
                    palette={
                      Object {
                        "backgroundColors": Object {
                          "button": "#3fbaca",
                          "common": "#ffffff",
                          "disabledButton": "#e2e2e2",
                          "main": "#f7f7f7",
                          "secondaryToggle": "#ffffff",
                          "toggle": "#425579",
                        },
                        "borderColors": Object {
                          "common": "#e2e2e2",
                        },
                        "fontColors": Object {
                          "anchor": "#3fbaca",
                          "button": "#ffffff",
                          "common": "#293346",
                          "description": "#9d9d9d",
                          "disabledAnchor": "#9d9d9d",
                          "label": "#2d2d2d",
                          "secondaryToggle": "#9d9d9d",
                          "text": "#2d2d2d",
                          "toggle": "#ffffff",
                        },
                      }
                    }
                    style={Object {}}
                    value="/preview"
                  >
                    <Box
                      active={false}
                      className="sc-dlnjPT cnwfAJ"
                      onClick={[Function]}
                      palette={
                        Object {
                          "backgroundColors": Object {
                            "button": "#3fbaca",
                            "common": "#ffffff",
                            "disabledButton": "#e2e2e2",
                            "main": "#f7f7f7",
                            "secondaryToggle": "#ffffff",
                            "toggle": "#425579",
                          },
                          "borderColors": Object {
                            "common": "#e2e2e2",
                          },
                          "fontColors": Object {
                            "anchor": "#3fbaca",
                            "button": "#ffffff",
                            "common": "#293346",
                            "description": "#9d9d9d",
                            "disabledAnchor": "#9d9d9d",
                            "label": "#2d2d2d",
                            "secondaryToggle": "#9d9d9d",
                            "text": "#2d2d2d",
                            "toggle": "#ffffff",
                          },
                        }
                      }
                      style={Object {}}
                      value="/preview"
                    >
                      <Component
                        active={false}
                        className="sc-dlnjPT cnwfAJ"
                        onClick={[Function]}
                        palette={
                          Object {
                            "backgroundColors": Object {
                              "button": "#3fbaca",
                              "common": "#ffffff",
                              "disabledButton": "#e2e2e2",
                              "main": "#f7f7f7",
                              "secondaryToggle": "#ffffff",
                              "toggle": "#425579",
                            },
                            "borderColors": Object {
                              "common": "#e2e2e2",
                            },
                            "fontColors": Object {
                              "anchor": "#3fbaca",
                              "button": "#ffffff",
                              "common": "#293346",
                              "description": "#9d9d9d",
                              "disabledAnchor": "#9d9d9d",
                              "label": "#2d2d2d",
                              "secondaryToggle": "#9d9d9d",
                              "text": "#2d2d2d",
                              "toggle": "#ffffff",
                            },
                          }
                        }
                        style={Object {}}
                        value="/preview"
                      >
                        <StyledBox
                          active={false}
                          className="sc-dlnjPT cnwfAJ"
                          onClick={[Function]}
                          palette={
                            Object {
                              "backgroundColors": Object {
                                "button": "#3fbaca",
                                "common": "#ffffff",
                                "disabledButton": "#e2e2e2",
                                "main": "#f7f7f7",
                                "secondaryToggle": "#ffffff",
                                "toggle": "#425579",
                              },
                              "borderColors": Object {
                                "common": "#e2e2e2",
                              },
                              "fontColors": Object {
                                "anchor": "#3fbaca",
                                "button": "#ffffff",
                                "common": "#293346",
                                "description": "#9d9d9d",
                                "disabledAnchor": "#9d9d9d",
                                "label": "#2d2d2d",
                                "secondaryToggle": "#9d9d9d",
                                "text": "#2d2d2d",
                                "toggle": "#ffffff",
                              },
                            }
                          }
                          style={Object {}}
                          value="/preview"
                        >
                          <div
                            className="sc-bdnylx sJSbd sc-dlnjPT cnwfAJ"
                            onClick={[Function]}
                            style={Object {}}
                            value="/preview"
                          >
                            Preview
                          </div>
                        </StyledBox>
                      </Component>
                    </Box>
                  </StyledSidemenuItem>
                </Component>
              </WithMooskinContext>
            </div>
          </StyledBox>
        </Component>
      </Box>
    </StyledSidemenu>
  </Component>
</WithMooskinContext>
`;

exports[`Sidemenu renders SidemenuItem correctly 1`] = `
<WithMooskinContext
  active={true}
  className="myClass"
  onClick={[MockFunction]}
  style={
    Object {
      "color": "blue",
    }
  }
>
  <Component
    active={true}
    className="myClass"
    onClick={[MockFunction]}
    palette={
      Object {
        "backgroundColors": Object {
          "button": "#3fbaca",
          "common": "#ffffff",
          "disabledButton": "#e2e2e2",
          "main": "#f7f7f7",
          "secondaryToggle": "#ffffff",
          "toggle": "#425579",
        },
        "borderColors": Object {
          "common": "#e2e2e2",
        },
        "fontColors": Object {
          "anchor": "#3fbaca",
          "button": "#ffffff",
          "common": "#293346",
          "description": "#9d9d9d",
          "disabledAnchor": "#9d9d9d",
          "label": "#2d2d2d",
          "secondaryToggle": "#9d9d9d",
          "text": "#2d2d2d",
          "toggle": "#ffffff",
        },
      }
    }
    style={
      Object {
        "color": "blue",
      }
    }
  >
    <StyledSidemenuItem
      active={true}
      className="myClass"
      onClick={[MockFunction]}
      palette={
        Object {
          "backgroundColors": Object {
            "button": "#3fbaca",
            "common": "#ffffff",
            "disabledButton": "#e2e2e2",
            "main": "#f7f7f7",
            "secondaryToggle": "#ffffff",
            "toggle": "#425579",
          },
          "borderColors": Object {
            "common": "#e2e2e2",
          },
          "fontColors": Object {
            "anchor": "#3fbaca",
            "button": "#ffffff",
            "common": "#293346",
            "description": "#9d9d9d",
            "disabledAnchor": "#9d9d9d",
            "label": "#2d2d2d",
            "secondaryToggle": "#9d9d9d",
            "text": "#2d2d2d",
            "toggle": "#ffffff",
          },
        }
      }
      style={
        Object {
          "color": "blue",
        }
      }
    >
      <Box
        active={true}
        className="sc-dlnjPT dhtinz myClass"
        onClick={[MockFunction]}
        palette={
          Object {
            "backgroundColors": Object {
              "button": "#3fbaca",
              "common": "#ffffff",
              "disabledButton": "#e2e2e2",
              "main": "#f7f7f7",
              "secondaryToggle": "#ffffff",
              "toggle": "#425579",
            },
            "borderColors": Object {
              "common": "#e2e2e2",
            },
            "fontColors": Object {
              "anchor": "#3fbaca",
              "button": "#ffffff",
              "common": "#293346",
              "description": "#9d9d9d",
              "disabledAnchor": "#9d9d9d",
              "label": "#2d2d2d",
              "secondaryToggle": "#9d9d9d",
              "text": "#2d2d2d",
              "toggle": "#ffffff",
            },
          }
        }
        style={
          Object {
            "color": "blue",
          }
        }
      >
        <Component
          active={true}
          className="sc-dlnjPT dhtinz myClass"
          onClick={[MockFunction]}
          palette={
            Object {
              "backgroundColors": Object {
                "button": "#3fbaca",
                "common": "#ffffff",
                "disabledButton": "#e2e2e2",
                "main": "#f7f7f7",
                "secondaryToggle": "#ffffff",
                "toggle": "#425579",
              },
              "borderColors": Object {
                "common": "#e2e2e2",
              },
              "fontColors": Object {
                "anchor": "#3fbaca",
                "button": "#ffffff",
                "common": "#293346",
                "description": "#9d9d9d",
                "disabledAnchor": "#9d9d9d",
                "label": "#2d2d2d",
                "secondaryToggle": "#9d9d9d",
                "text": "#2d2d2d",
                "toggle": "#ffffff",
              },
            }
          }
          style={
            Object {
              "color": "blue",
            }
          }
        >
          <StyledBox
            active={true}
            className="sc-dlnjPT dhtinz myClass"
            onClick={[MockFunction]}
            palette={
              Object {
                "backgroundColors": Object {
                  "button": "#3fbaca",
                  "common": "#ffffff",
                  "disabledButton": "#e2e2e2",
                  "main": "#f7f7f7",
                  "secondaryToggle": "#ffffff",
                  "toggle": "#425579",
                },
                "borderColors": Object {
                  "common": "#e2e2e2",
                },
                "fontColors": Object {
                  "anchor": "#3fbaca",
                  "button": "#ffffff",
                  "common": "#293346",
                  "description": "#9d9d9d",
                  "disabledAnchor": "#9d9d9d",
                  "label": "#2d2d2d",
                  "secondaryToggle": "#9d9d9d",
                  "text": "#2d2d2d",
                  "toggle": "#ffffff",
                },
              }
            }
            style={
              Object {
                "color": "blue",
              }
            }
          >
            <div
              className="sc-bdnylx sJSbd sc-dlnjPT dhtinz myClass"
              onClick={[MockFunction]}
              style={
                Object {
                  "color": "blue",
                }
              }
            />
          </StyledBox>
        </Component>
      </Box>
    </StyledSidemenuItem>
  </Component>
</WithMooskinContext>
`;
